executor/common_bsd.h

```diff
@@ -41,6 +72,212 @@ static uintptr_t syz_open_pts(void)
 	return slave;
 }
 
+#endif  // SYZ_EXECUTOR || __NR_syz_open_pts
+
+#if SYZ_EXECUTOR || SYZ_TUN_ENABLE
+
+#include <fcntl.h>
+#include <sys/types.h>
+#include <net/if_tun.h>
+
+static int tunfd = -1;
+
+// We just need this to be large enough to hold headers that we parse (ethernet/ip/tcp).
+// Rest of the packet (if any) will be silently truncated which is fine.
+#define SYZ_TUN_MAX_PACKET_SIZE 1000
+
+#define TUN_IFACE "tap0"
+#define TUN_DEVICE "/dev/tap0"
+
+#define LOCAL_IPV4 "172.20.20.170"
+#define LOCAL_IPV6 "fe80::aa"
+
+static void initialize_tun(void)
+{
+#if SYZ_EXECUTOR
+	if (!flag_enable_tun)
+		return;
+#endif  // SYZ_EXECUTOR
+	tunfd = open(TUN_DEVICE, O_RDWR | O_NONBLOCK);
+	if (tunfd == -1) {
+#if SYZ_EXECUTOR
+	        fail("tun: can't open %s\n", TUN_DEVICE);
+#else
+		printf("tun: can't open %s: errno=%d\n", TUN_DEVICE, errno);
+		return;
+#endif  // SYZ_EXECUTOR
+	}
+	// Remap tun onto higher fd number to hide it from fuzzer and to keep
+	// fd numbers stable regardless of whether tun is opened or not (also see kMaxFd).
+	const int kTunFd = 240;
+	if (dup2(tunfd, kTunFd) < 0)
+		fail("dup2(tunfd, kTunFd) failed");
+	close(tunfd);
+	tunfd = kTunFd;
+
+	execute_command(1, "ifconfig %s inet %s", TUN_IFACE, LOCAL_IPV4);
```

Comment:
A less hacky way would be to use the `SIOCAIFADDR` ioctl as described in netintro(4). But this might as well be fine since it's only done during init phase.

Reply:
If we rely on rdomain(4) for isolation this may become a less clear win considering the number of different commands than will need to be reimplemented in C, WDYT?